### DISCUSSION TITLE
[question] How does the front-end obtain data from environmental variables?

### DISCUSSION THREAD

**Discussion by JinSooo** [tauri-apps/tauri#8238] [Q&A] [answered]
I want to obtain data from environmental variables.
```env
NEXT_PUBLIC_DB_HOST=
NEXT_PUBLIC_DB_USER=
NEXT_PUBLIC_DB_PASS=
NEXT_PUBLIC_DB_DATABASE=
```
and then use it at front-end
```typescript
Database.load(`mysql://${process.env.NEXT_PUBLIC_DB_USER}:${process.env.NEXT_PUBLIC_DB_PASS}@${process.env.NEXT_PUBLIC_DB_HOST}/${process.env.NEXT_PUBLIC_DB_DATABASE}`)
```

**Comment by FabianLars** [comment, marked as answer]
So in development this is up to your frontend bundler (vitejs, webpack or whatever). This is also the case if you want to set these env vars on your build system, and have them hardcoded in your frontend files.

If you want to read them at runtime from the user's env then it will be tricky because your frontend runs in the browser's sandbox which means you'll have to read these env vars on the rust side instead.
